core/planner.cpp

```diff
@@ -29,18 +29,19 @@
 
 static constexpr int base_timestep = 2; // seconds
 
-static int decostoplevels_metric[] = { 0, 3000, 6000, 9000, 12000, 15000, 18000, 21000, 24000, 27000,
-					30000, 33000, 36000, 39000, 42000, 45000, 48000, 51000, 54000, 57000,
-					60000, 63000, 66000, 69000, 72000, 75000, 78000, 81000, 84000, 87000,
-					90000, 100000, 110000, 120000, 130000, 140000, 150000, 160000, 170000,
-					180000, 190000, 200000, 220000, 240000, 260000, 280000, 300000,
-					320000, 340000, 360000, 380000 };
-static int decostoplevels_imperial[] = { 0, 3048, 6096, 9144, 12192, 15240, 18288, 21336, 24384, 27432,
-					30480, 33528, 36576, 39624, 42672, 45720, 48768, 51816, 54864, 57912,
-					60960, 64008, 67056, 70104, 73152, 76200, 79248, 82296, 85344, 88392,
-					91440, 101600, 111760, 121920, 132080, 142240, 152400, 162560, 172720,
-					182880, 193040, 203200, 223520, 243840, 264160, 284480, 304800,
-					325120, 345440, 365760, 386080 };
+static std::vector<depth_t> decostoplevels_metric = { 0_m, 3_m, 6_m, 9_m, 12_m, 15_m, 18_m, 21_m, 24_m, 27_m,
+					30_m, 33_m, 36_m, 39_m, 42_m, 45_m, 48_m, 51_m, 54_m, 57_m,
+					60_m, 63_m, 66_m, 69_m, 72_m, 75_m, 78_m, 81_m, 84_m, 87_m,
+					90_m, 100_m, 110_m, 120_m, 130_m, 140_m, 150_m, 160_m, 170_m,
+					180_m, 190_m, 200_m, 220_m, 240_m, 260_m, 280_m, 300_m,
+					320_m, 340_m, 360_m, 380_m };
+// Note: for fractional feet (e.g. 333.33 ft), we use mm, since currently our custom depth literals only support integers.
+static std::vector<depth_t> decostoplevels_imperial = { 0_ft, 10_ft, 20_ft, 30_ft, 40_ft, 50_ft, 60_ft, 70_ft, 80_ft, 90_ft,
```

Comment:
Frankly, not sure if we want to keep these in fractions, or if we should just truncate to 333 ft, 666 ft and so on - these are mostly theoretical depths anyway as there's only a handful of people on this planet diving to depths that will require stops at these depths.